File: Easy/Max Sum Subarray of size K/max-sum-subarray-of-size-k.cpp
//{ Driver Code Starts
#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
class Solution{   
public:
    long maximumSumSubarray(int k, vector<int> &arr , int N){
        // code here 
        int i=0,j=0;
        long long sum=0,maxy=-99999;
        while(j<N)
        {
            if(j-i+1<k)
            {
                sum+=arr[j];
                  maxy=max(maxy,sum);
                  j++;
            }
            else if(j-i+1==k)
            {
                sum+=arr[j];
                maxy=max(maxy,sum);
                sum-=arr[i];
                j++;
                i++;
                
            }
        }
        return maxy;
    }
};

//{ Driver Code Starts.
int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        int N,K;
        cin >> N >> K;;
        vector<int>Arr;
        for(int i=0;i<N;++i){
            int x;
            cin>>x;
            Arr.push_back(x);
        }
        Solution ob;
        cout << ob.maximumSumSubarray(K,Arr,N) << endl;
    }
    return 0; 
} 
// } Driver Code Ends

File: Difficulty: Easy/Max Sum Subarray of size K/max-sum-subarray-of-size-k.java
class Solution {
    public int maxSubarraySum(int[] arr, int k) {
        // Code here
                int sum = 0;
        int i = 0;
        int j = 0;
        int n = arr.length;
        int ans = Integer.MIN_VALUE;
        while (j < n) {
            sum += arr[j];
            if ((j - i + 1) < k) {
                // continue;
            } else if ((j - i + 1) == k) {
                ans = Math.max(sum, ans);
                sum = sum - arr[i];
                i++;
            }
            j++;
        }
        return ans;
    }
}